solvcon/mesh.pyx

```diff
@@ -221,6 +221,17 @@ cdef class Mesh:
             if vals[0] != vals[1]:
                 msgs.append('body(%d,%d)' % vals)
             if msgs:
+                f = open("/tmp/solvcon.debug.log", "a")
+                msgs.append("\n" + str(dir(_table)) + "\n")
+                msgs.append(str(_cnda))
+                msgs.append(str(_table._ghostaddr))
+                msgs.append(str(_table._bodyaddr))
+                msgs.append(str(_table.nghost))
+                msgs.append(str(_table.nbody))
+                #msgs.append(str(_table._full))
+                #msgs.append(str(_table._body))
+                f.write(','.join(msgs) + "\n")
+                f.close()
                 tmpl = '%s array mismatch: %s'
```

Comment:
this debug code could reproduce similar error message from issue#135 and it looks like:



```
└──╼ nosetests test_builtin.py:TestLocalParallel.test_soln
/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/io/netcdf.py:47: RuntimeWarning: netCDF4 isn't built; No module named 'netCDF4'
  sc.import_module_may_fail('netCDF4')
Process Process-1:
Process Process-2:
Process Process-3:
Traceback (most recent call last):
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 209, in run
    self.eventloop()
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 163, in eventloop
    self._eventloop()
Traceback (most recent call last):
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 142, in _eventloop
    ret = method(*ntc.args, **ntc.kw)
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/solver_legacy.py", line 293, in check_blk_pointer
    msh._check_pointers(self.blk)
  File "solvcon/mesh.pyx", line 236, in solvcon.mesh.Mesh._check_pointers (solvcon/mesh.c:4748)
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
AttributeError: ndcrd array mismatch: shared(5,43002016), body(0,42978496), 
['B', 'F', 'G', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattr__', '__getattribute__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '_body', '_bodyaddr', '_bodypart', '_calc_offset', '_full', '_ghostaddr', '_ghostpart', '_nda', 'nbody', 'nghost', 'offset']
, None, 5, 0, -1, -1
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 209, in run
    self.eventloop()
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 163, in eventloop
    self._eventloop()
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 142, in _eventloop
    ret = method(*ntc.args, **ntc.kw)
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/solver_legacy.py", line 293, in check_blk_pointer
    msh._check_pointers(self.blk)
  File "solvcon/mesh.pyx", line 236, in solvcon.mesh.Mesh._check_pointers (solvcon/mesh.c:4748)
AttributeError: ndcrd array mismatch: shared(5,43005840), body(0,42981120), 
['B', 'F', 'G', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattr__', '__getattribute__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '_body', '_bodyaddr', '_bodypart', '_calc_offset', '_full', '_ghostaddr', '_ghostpart', '_nda', 'nbody', 'nghost', 'offset']
, None, 5, 0, -1, -1
Traceback (most recent call last):
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 209, in run
    self.eventloop()
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 163, in eventloop
    self._eventloop()
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 142, in _eventloop
    ret = method(*ntc.args, **ntc.kw)
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/solver_legacy.py", line 293, in check_blk_pointer
    msh._check_pointers(self.blk)
  File "solvcon/mesh.pyx", line 236, in solvcon.mesh.Mesh._check_pointers (solvcon/mesh.c:4748)
AttributeError: ndcrd array mismatch: shared(5,43020448), body(0,42990896), 
['B', 'F', 'G', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattr__', '__getattribute__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '_body', '_bodyaddr', '_bodypart', '_calc_offset', '_full', '_ghostaddr', '_ghostpart', '_nda', 'nbody', 'nghost', 'offset']
, None, 5, 0, -1, -1
E
======================================================================
ERROR: test_soln (test_builtin.TestLocalParallel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tai271828/work-my-projects/solvcon/branch.160523.issue135-16w11-01/ftests/parallel/test_builtin.py", line 166, in test_soln
    case = self._get_case(npart=self.npart, domaintype=Collective)
  File "/home/tai271828/work-my-projects/solvcon/branch.160523.issue135-16w11-01/ftests/parallel/test_builtin.py", line 121, in _get_case
    case.init()
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/case_legacy.py", line 483, in init
    self._remote_init_solver()
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/case_legacy.py", line 639, in _remote_init_solver
    dealer.barrier()
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 586, in barrier
    assert issubclass(sdw.recv(), Barrier)
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/rpc.py", line 434, in recv
    return self.conn.recv(*args, **kw)
  File "/home/tai271828/work-my-projects/solvcon/solvcon.installed/solvcon/connection.py", line 152, in recv
    return self.conn.recv(*args, **kw)
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/connection.py", line 250, in recv
    buf = self._recv_bytes()
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/connection.py", line 407, in _recv_bytes
    buf = self._recv(4)
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/multiprocessing/connection.py", line 379, in _recv
    chunk = read(handle, remaining)
ConnectionResetError: [Errno 104] Connection reset by peer
-------------------- >> begin captured stdout << ---------------------
********************************************************************************
*** Start init (level 0) blockcase ... 
*** ****************************************************************************
*** ****************************************************************************
*** *** Start build_domain ... 
*** *** ************************************************************************
*** *** mesh file: /home/tai271828/work-my-projects/solvcon/solvcon.installed/test/data/sample.neu
*** *** ************************************************************************
*** *** *** Start read_neu_data from /home/tai271828/work-my-projects/solvcon/solvcon.installed/test/data/sample.neu ... 
*** *** *** ********************************************************************
*** *** *** ********************************************************************
*** *** *** End read_neu_data . Elapsed time (sec) = 0.000125408
*** *** ************************************************************************
*** *** ************************************************************************
*** *** *** Start create_neu_object ... 
*** *** *** ********************************************************************
*** *** *** ********************************************************************
*** *** *** End create_neu_object . Elapsed time (sec) = 0.00238514
*** *** ************************************************************************
*** *** ************************************************************************
*** *** *** Start convert_neu_to_block ... 
*** *** *** ********************************************************************
*** *** *** ********************************************************************
*** *** *** End convert_neu_to_block . Elapsed time (sec) = 0.00422168
*** *** ************************************************************************
*** *** ************************************************************************
*** *** End build_domain . Elapsed time (sec) = 0.00704241
*** ****************************************************************************
*** 
*** ****************************************************************************
*** *** Start split_domain ... 
*** *** ************************************************************************
*** *** ************************************************************************
*** *** End split_domain . Elapsed time (sec) = 0.0123496
*** ****************************************************************************
*** 
*** ****************************************************************************
*** *** Start build_dealer ... 
*** *** ************************************************************************
*** *** ************************************************************************
*** *** End build_dealer . Elapsed time (sec) = 0.32678
*** ****************************************************************************
*** 
*** ****************************************************************************
*** *** Start interconnect ... 
*** *** ************************************************************************
*** *** 0 -> 1 2.
*** *** 1 -> 2.
*** *** ************************************************************************
*** *** End interconnect . Elapsed time (sec) = 0.309813
*** ****************************************************************************
*** 
*** ****************************************************************************
*** *** Start remote_init_solver ... 
*** *** ************************************************************************
*** *** solver #0/(3-1): sending ... done.
*** *** solver #1/(3-1): sending ... done.
*** *** solver #2/(3-1): sending ... done.
*** *** Bind/Init ... validation of block object adderess ... 
--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 1 test in 0.672s

FAILED (errors=1)
```